lib/wallet/src/Cardano/Wallet/DB/Store/Checkpoints.hs

```diff
@@ -237,6 +244,15 @@ mkStoreWallet wid =
     update1 (UpdateCheckpoints delta) =
         -- FIXME LATER during ADP-1043: remove 'undefined'
         updateS storeCheckpoints undefined delta
+    update1 (UpdateSubmissions delta) =
+        for_ (reverse delta) (updateReadS submissions)
+
+updateReadS :: Monad m => Store m da -> da -> m (Either SomeException ())
+updateReadS x d = do
+    ey <- loadS x
+    case ey of
+        Left e -> pure $ Left  e
+        Right y -> Right <$> updateS x (Just y) d
```

Comment:
```suggestion
        for_ (reverse delta) (updateS submissions Nothing)
```

We have `Nothing` now, so the `Store` will have to take care of loading the value if necessary.

However — we do have the value of type `TxSubmissions` available in memory as part of the `WalletState`, we can extract it and pass `Just` here. In contrast, by using `Nothing`, we will incur a performance penalty when updating the submissions (e.g. during rollback or pruning).

I suggest to proceed as follows: Use `Nothing` for this PR, and implement `Just` in a subsequent PR.